packages/walrus/src/storage-node/client.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { BlobMetadata, BlobMetadataWithId, SliverData } from '../utils/bcs.js';
+import { StorageNodeAPIError, UserAbortError } from './error.js';
+import type {
+	GetBlobMetadataRequestInput,
+	GetBlobMetadataResponse,
+	GetDeletableBlobConfirmationRequestInput,
+	GetDeletableBlobConfirmationResponse,
+	GetPermanentBlobConfirmationRequestInput,
+	GetPermanentBlobConfirmationResponse,
+	GetSliverRequestInput,
+	GetSliverResponse,
+	StoreBlobMetadataRequestInput,
+	StoreBlobMetadataResponse,
+	StoreSliverRequestInput,
+	StoreSliverResponse,
+} from './types.js';
+
+export type Fetch = (url: RequestInfo, init?: RequestInit) => Promise<Response>;
+
+export type StorageNodeClientOptions = {
+	/**
+	 * An optional custom fetch function.
+	 *
+	 * If not provided, defaults to the global `fetch` function (`globalThis.fetch`).
+	 *
+	 * @default globalThis.fetch
+	 */
+	fetch?: Fetch;
```

Comment:
We talked about making this actually `(request: Request) => Response` instead so that we can eventually shove in extra params